### PR TITLE
Install missing OCM components

### PR DESCRIPTION
**Pull request by nirs**
This adds the missing OCM components:
- multicloud-operators-foundation
- multicloud-operators-subscription
- governance policy framework

And fixes issues in cluster-manager and klusterlet, revealed by installing 
the subscription operator.

There is no self test for any of the new components, we can add this later.

### CODE REVIEW

**Review by raghavendra-talur — APPROVED** [review]

**Comment by raghavendra-talur** [comment]
Thanks, this looks good.

Also thanks for create #595. I believe fixing that would make the install scripts cleaner.

File: test/policy-framework/start [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            )
+            drenv.log_progress(
+                f"Waiting until {deploy} is rolled out in cluster {cluster}",
+            )
+            drenv.kubectl(
+                "rollout", "status", "deploy", deploy,
+                "--namespace", CLUSTER_NAMESPACE,
+                "--timeout", "300s",
+                profile=cluster,
+            )
```
Comment (nirs) [inline test/policy-framework/start:77]: Waiting until deployment exists and then waiting for rollout is done many times. Maybe
we need a higher level helper like wait_for() to do both.

File: test/policy-framework/start [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+def deploy_hub(hub):
+    drenv.log_progress(
+        f"Installing governance policy framework in cluster {hub}",
+    )
+    drenv.run(
+        "clusteradm", "install", "hub-addon",
+        "--names", "governance-policy-framework",
+        "--context", hub,
+    )
```
Comment (nirs) [inline test/policy-framework/start:22]: logging and installing with clusteradm is repeated here, and will be repeated in 
many scripts. I think we need a clusteradm wrapper module to make this nicer to use.

Something like:

    clusteadm.install(command, names, cluster)
    clusteradm.addon(command, name, clusters)

File: test/subscription-operator/common.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  - $base_url/deploy/common/apps.open-cluster-management.io_subscriptions.yaml
+  - $base_url/deploy/common/clusterrole.yaml
+  - $base_url/deploy/common/clusterrole_binding.yaml
+  - $base_url/deploy/common/namespace.yaml
+  - $base_url/deploy/common/service.yaml
+  - $base_url/deploy/common/service_account.yaml
+
+
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
```
Comment (nirs) [inline test/subscription-operator/common.yaml:19]: Can be eliminated by using clusteradm.

File: test/subscription-operator/hub.yaml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  - target:
+      group: apps
+      version: v1
+      kind: Deployment
+      name: multicluster-operators-subscription
+      namespace: multicluster-operators
+    patch: |-
+      - op: replace
+        path: /spec/template/spec/containers/0/image
+        value: quay.io/open-cluster-management/multicluster-operators-subscription:$image_tag
```
Comment (nirs) [inline test/subscription-operator/hub.yaml:18]: Can be eliminated by using clusteradm.

File: test/subscription-operator/managed.yaml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    patch: |-
+      - op: replace
+        path: /spec/template/spec/containers/0/image
+        value: quay.io/open-cluster-management/multicluster-operators-subscription:$image_tag
+      - op: replace
+        path: /spec/template/spec/containers/0/command/2
+        value: --cluster-name=$cluster_name
+      - op: replace
+        path: /spec/template/spec/containers/0/command/3
+        value: --cluster-namespace=$cluster_name
```
Comment (nirs) [inline test/subscription-operator/managed.yaml:23]: Can be eliminated by using clusteradm.

File: test/subscription-operator/start [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+for cluster in cluster1, cluster2:
+    drenv.log_progress(f"Deploying managed cluster operators in cluster {cluster}")
+    with drenv.kustomization(
+        f"subscription-operator/managed.yaml",
+        base_url=BASE_URL,
+        image_tag=IMAGE_TAG,
+        cluster_name=cluster,
+    ) as kustomization:
+        drenv.kubectl("apply", "--kustomize", kustomization, profile=cluster)
```
Comment (nirs) [inline test/subscription-operator/start:84]: Everything above can be replaced with simpler clusteradm commands installing latest
release.

File: test/ocm-controller/kustomization.yaml [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+- name: quay.io/stolostron/multicloud-manager
+  newName: quay.io/stolostron/multicloud-manager
+  newTag: latest
+
+
+patchesStrategicMerge:
+- $base_url/patches.yaml
+
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
```
Comment (nirs) [inline test/ocm-controller/kustomization.yaml:34]: Can be eliminated by using clusteradm.

File: test/ocm-controller/start [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    input=yaml.safe_dump(namespace),
+    profile=cluster,
+)
+
+drenv.log_progress(f"Deploying ocm controller in cluster {cluster}")
+with drenv.kustomization(
+    "ocm-controller/kustomization.yaml",
+    base_url=BASE_URL,
+) as kustomization:
+    drenv.kubectl("apply", "--kustomize", kustomization, profile=cluster)
```
Comment (nirs) [inline test/ocm-controller/start:43]: Code above can be replaced with one clusteradm comamnd.

**Review by nirs — COMMENTED** [review]
Add comments for future changes.

**Comment by ShyamsundarR** [comment]
Comments added here are for future work I assume, would that be right @nirs ?

Also, can clusteradm be used to install a specific version than latest? This would be important to ensure we test our bits on a stable stack than the latest from clusteradm.

**Comment by nirs** [comment]
> Comments added here are for future work I assume, would that be right @nirs ?

Yes, there is only one trivial change (use a loop) that I can do now. If there is
a need for a new version I'll do this now.

> Also, can clusteradm be used to install a specific version than latest? This would be important to ensure we test our bits on a stable stack than the latest from clusteradm.

How granular do we need to be? Do we need specific version of OCM (e.g. ACM release)
or specific version of OCM component (e.g. policy framework)?

I think we should limit the flexibility since supporting multiple versions is expensive.
The main purpose of the testing environment is to test with latest upstream release to 
find issues early, so when latest release reach downstream ramen will be ready. Testing
ramen with downstream should be done by the downstream folks (QE).

Anyway I will check about clusteadm version support.

**Comment by nirs** [comment]
clusteradm init (not used yet) has:

    --bundle-version='default': the version of predefined compatible image versions.
    e.g. v0.6.0, defaulted to the latest release version. also, we can set "latest"
    to install latest develop version

So it looks like we can control one OCM bundle version, which affects all components.

@ShyamsundarR do you think this is good enough?

**Comment by ShyamsundarR** [comment]
> clusteradm init (not used yet) has:
> 
> ```
> --bundle-version='default': the version of predefined compatible image versions.
> e.g. v0.6.0, defaulted to the latest release version. also, we can set "latest"
> to install latest develop version
> ```
> 
> So it looks like we can control one OCM bundle version, which affects all components.
> 
> @ShyamsundarR do you think this is good enough?

That is enough, thanks!

**Pull request merged**